nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/lossfunctions/impl/LossMAE.java

```diff
@@ -67,6 +72,12 @@ public INDArray computeGradient(INDArray labels, INDArray preOutput, IActivation
         return gradients;
     }
 
+    @Override
+    public @NonNull SDVariable defineLoss(@NonNull SameDiff sameDiff, @NonNull SDVariable input,
+            @NonNull SDVariable labels) {
+        return LossUtil.batchAverage(defineFullLossArray(sameDiff, input, labels).div(labels.shape().get(SDIndex.point(1))));
```

Comment:
I'm not following the `.get(SDIndex.point(1))` part. Is that effectively a reshape to rank 0 scalar or something?

Reply:
It's essentially replicating `mean(true, 1)`, for some reason mean was causing issues.  Enough has changed since then that I'll try it again.